### PR TITLE
Feature/std ranges part8

### PR DESCRIPTION
**Pull request by whaeck**
Adding take, drop, take_while and drop_while views.

### CODE REVIEW

File: src/tools/std20/algorithm/test/find.test.cpp [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+// what we are testing
+#include "tools/std20/algorithm.hpp"
+
+// other includes
+
+// convenience typedefs
+//using namespace njoy::tools;
+namespace std20 = nano;
+
+SCENARIO( "equal" ) {
```
Comment (mswwilson) [inline src/tools/std20/algorithm/test/find.test.cpp:13]: "equal" -> "find"

Comment (whaeck) [reply src/tools/std20/algorithm/test/find.test.cpp:13]: addressed in fix/review

File: src/tools/std20/views/test/drop.test.cpp [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+		    CHECK( std20::ranges::view< Range > );
+		    CHECK( ! std20::ranges::sized_range < Range > );
+        CHECK( std20::ranges::forward_range< Range > );
+        CHECK( ! std20::ranges::bidirectional_range< Range > );
+		    CHECK( ! std20::ranges::random_access_range < Range > );
+		    CHECK( ! std20::ranges::contiguous_range < Range > );
+		    CHECK( std20::ranges::common_range < Range > );
+      }
+
+      THEN( "the take_view range and iterator associated types are correct" ) {
```
Comment (mswwilson) [inline src/tools/std20/views/test/drop.test.cpp:44]: "take_view" -> "drop" 

Likewise for other THENs in scenario

File: src/tools/std20/views/test/drop_while.test.cpp [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+		    CHECK( std20::ranges::view< Range > );
+		    CHECK( ! std20::ranges::sized_range < Range > );
+        CHECK( std20::ranges::forward_range< Range > );
+        CHECK( ! std20::ranges::bidirectional_range< Range > );
+		    CHECK( ! std20::ranges::random_access_range < Range > );
+		    CHECK( ! std20::ranges::contiguous_range < Range > );
+		    CHECK( std20::ranges::common_range < Range > );
+      }
+
+      THEN( "the take_view range and iterator associated types are correct" ) {
```
Comment (mswwilson) [inline src/tools/std20/views/test/drop_while.test.cpp:44]: "take_view" -> "drop_while" 

Likewise for other THENs in scenario

File: src/tools/std20/views/test/join.test.cpp [hunk 4]
```diff
@@ -0,0 +1,183 @@
...
+		    CHECK( std20::ranges::view< Range > );
+		    CHECK( ! std20::ranges::sized_range < Range > );
+        CHECK( std20::ranges::forward_range< Range > );
+        CHECK( ! std20::ranges::bidirectional_range< Range > );
+		    CHECK( ! std20::ranges::random_access_range < Range > );
+		    CHECK( ! std20::ranges::contiguous_range < Range > );
+		    CHECK( std20::ranges::common_range < Range > );
+      }
+
+      THEN( "the all_view range and iterator associated types are correct" ) {
```
Comment (mswwilson) [inline src/tools/std20/views/test/join.test.cpp:43]: "all_view" -> "join" 

Likewise for other THENs in scenario

File: src/tools/std20/views/test/single.test.cpp [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+//using namespace njoy::tools;
+namespace std20 = nano;
+
+SCENARIO( "single_view" ) {
+
+  auto chunk = std20::views::single( 3 );
+  using Range = decltype(chunk);
+  using Iterator = nano::iterator_t< Range >;
+
+  THEN( "the take_view satisfies the required concepts" ) {
```
Comment (mswwilson) [inline src/tools/std20/views/test/single.test.cpp:19]: "take_view" -> "single" 

Likewise for other THENs in scenario

File: src/tools/std20/views/test/take_while.test.cpp [hunk 6]
```diff
@@ -0,0 +1,172 @@
...
+
+    std::forward_list< int > values = { -2, -1, 0, 1, 2 };
+
+    WHEN( "when iterators are used" ) {
+
+      auto chunk = values | std20::views::take_while( predicate );
+      using Range = decltype(chunk);
+      using Iterator = nano::iterator_t< Range >;
+
+      THEN( "the take_view satisfies the required concepts" ) {
```
Comment (mswwilson) [inline src/tools/std20/views/test/take_while.test.cpp:33]: "take_view" -> "take_while" 

Likewise for other THENs in scenario

**Review by mswwilson — DISMISSED** [review]

File: src/tools/std20/views/test/drop.test.cpp (hunk 2 above)
Comment (whaeck) [reply src/tools/std20/views/test/drop.test.cpp:44]: fixed in fix/review

File: src/tools/std20/views/test/drop_while.test.cpp (hunk 3 above)
Comment (whaeck) [reply src/tools/std20/views/test/drop_while.test.cpp:44]: same

File: src/tools/std20/views/test/join.test.cpp (hunk 4 above)
Comment (whaeck) [reply src/tools/std20/views/test/join.test.cpp:43]: same

File: src/tools/std20/views/test/single.test.cpp (hunk 5 above)
Comment (whaeck) [reply src/tools/std20/views/test/single.test.cpp:19]: same

File: src/tools/std20/views/test/take_while.test.cpp (hunk 6 above)
Comment (whaeck) [reply src/tools/std20/views/test/take_while.test.cpp:33]: same

**Pull request merged**